### PR TITLE
New data models for hotkeys and store the key bindings in settings

### PR DESCRIPTION
**Pull request by rickychandra**
Partially closes: #1162 
This PR is the first one to enable customizable key bindings feature. I plan to submit 1-2 more PRs, so that it will be easier to review, as there could be many additions and changes.

In this PR:
- New data models for hotkeys, with the separation of key combinations by OSes (mac, win, and linux). The models and namings are loosely inspired by those of VS Code's.
- Save key bindings in DB, in `settings` model.
- Give each hotkeys an `id`, the style is similar to that of VS Code's; except that this one uses `_` instead of `.`, e.g., `request_showSettings` instead of `request.showSettings`. The reason is that NeDB prohibits naming fields with `.`.
- Use `⌥` symbol for Mac's option key, previously was `^` (is this intended?).
- Some codes' styles are changing, like [this](https://github.com/getinsomnia/insomnia/compare/develop...rickychandra:feature/custom-keybindings?expand=1#diff-3ce6983a68cc85e7b663f4847a5bb00dR479). I didn't intend to do this, maybe it's the git hook's prettifier work.

Actually I have only tested on Linux, because I couldn't get my hands on MacOS. :grimacing:

### CODE REVIEW

**Comment by Gabb1995** [comment]
Does this fix #685  ?

**Comment by rickychandra** [comment]
@Gabb1995 If this PR's final feature has been fully implemented, then yes, it can be used to remove the ctrl+d keybinding. But this PR is only the first step. :smile:

**Comment by gschier** [comment]
Hey @rickychandra. Sorry, I just haven't had time to take a look yet. I have a task next week to get all the open PRs dealt with so I'll let you know 😃

File: packages/insomnia-app/app/common/hotkeys.js [hunk 1]
```diff
@@ -1,305 +1,378 @@
...
+  keyCode: number,
 };
 
-export const SHOW_WORKSPACE_SETTINGS: Hotkey = {
-  description: 'Show Workspace Settings',
-  meta: true,
-  alt: false,
-  shift: true,
-  keycode: keycodes.comma,
+type GenericKeyCombination = {
```
[the remark this answers is not in the document]
Comment (rickychandra) [reply packages/insomnia-app/app/common/hotkeys.js]: Sure, I also think it's simpler without it. My first intention is actually to make each hotkey registration takes less line of code, but it seems that it doesn't matter that much.

**Comment by rickychandra** [comment]
Hi @gschier, any update on this? I have updated the codes following your comments. :smile:

**Comment by gschier** [comment]
Will try and get to it this week! I've been focusing on a few other things lately but am getting back into app stuff now

**Pull request merged**